hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -185,6 +187,8 @@
 
     private volatile boolean joinInProgress;
 
+    private final ExecutorService membershipEventExecutor = Executors.newSingleThreadExecutor();
```

Comment:
rather than creating a standalone `ExecutorService`, we can register one via out `ExecutionService`

```
private static final String MEMBERSHIP_EVENT_EXECUTOR_NAME = "hz:cluster:event";
```
and in `init` method:
```
executionService.register(MEMBERSHIP_EVENT_EXECUTOR_NAME, 1, Integer.MAX_VALUE, ExecutorType.CACHED);
```
and the usage:
```
nodeEngine.getExecutionService().execute(MEMBERSHIP_EVENT_EXECUTOR_NAME, new Runnable() {
                    public void run() {
                        if (added) {
                            service.memberAdded(event);
                        } else {
                            service.memberRemoved(event);
                        }
                    }
                });
```